### ISSUE TITLE
Consider discarding original value snapshot after use in full notification change tracking

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#730]
When full-notification change tracking is used a snapshot of the original value is taken in the Changing event which is then used in the Changed event. This snapshot value persists on the state entry even though it may not now be needed, so it could potentially be discarded once Changed is done with it. However, there is some value in keeping it around for situations such as: https://entityframework.codeplex.com/workitem/2435

**Comment by ajcvickers** [comment]
Closing this. Some original values need to be kept around so that the update pipeline can make use of them. This means that we can not, in general, discard original values. Once we keep any original value around there is no space penalty for keeping them all since they are allocated as a block. Therefore, given the way the design has evolved, it really doesn't make sense to attempt to do this.
